### PR TITLE
Document `substringUTF8` and `substringIndexUTF8`

### LINKED ISSUE #2016: Document `substringXYZ` functions.
Create documentation for the `substringXYZ` function. See #1833 for details on how to complete this issue.

- [ ] substringIndexUTF8
- [ ] substringUTF8

Child issue of #1833.

### PR DESCRIPTION
**Pull request by johnnymatthews**
### Changelog category (leave one):

- Documentation (changelog entry is not required)

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

---

Closes https://github.com/ClickHouse/clickhouse-docs/issues/2016

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 2196c75dd8ddaeb1d2f18ca7b05fb4ae37550a4b with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/61101/2196c75dd8ddaeb1d2f18ca7b05fb4ae37550a4b/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>❌ failure</td></tr>
<tbody>
</table>

File: docs/en/sql-reference/functions/string-functions.md [hunk 1]
```diff
@@ -588,8 +588,41 @@ Result:
...
+
+A substring of `s` with `length` many bytes, starting at index `offset`.
+
+**Implementation details**
+
+Assumes that the string contains valid UTF-8 encoded text. If this assumption is violated, no exception is thrown and the result is undefined.
+
+**Example**
+
+```sql
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/string-functions.md:617]: Let's use an example where UTF8 support actually makes a difference. What about this:

```sql
SELECT 'Täglich grüßt das Murmeltier.' AS str,
       substringUTF8(str, 9),
       substringUTF8(str, 9, 5)
FORMAT Vertical
```

(`ü` and `ß` take up two bytes each)

Result: https://fiddle.clickhouse.com/f968cc24-0d42-49fd-929b-97674040e1ad

File: docs/en/sql-reference/functions/string-functions.md [hunk 2]
```diff
@@ -624,7 +657,39 @@ Result:
...
+A substring [String](../../sql-reference/data-types/string.md) of `s` before `count` occurrences of `delim`.
+
+**Implementation details**
+
+Assumes that the string contains valid UTF-8 encoded text. If this assumption is violated, no exception is thrown and the result is undefined.
+
+**Example**
+
+```sql
+SELECT substringIndexUTF8('www.clickhouse.com', '.', 2)
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/string-functions.md:687]: Same here. A domain name is okay, but we should use one with "true" Unicode codepoints (that's actually also possible via internationalized domain names). Maybe `www.straßen-in-europa.de` ("streets-in-europe")?

File: docs/en/sql-reference/functions/string-functions.md [hunk 3]
```diff
@@ -588,8 +588,41 @@ Result:
...
+
+**Implementation details**
+
+Assumes that the string contains valid UTF-8 encoded text. If this assumption is violated, no exception is thrown and the result is undefined.
+
+**Example**
+
+```sql
+SELECT 'database' AS string, substringUTF8(string, 5), substringUTF8(string, 5, 1)
+```
```
Comment (johnnymatthews) [inline docs/en/sql-reference/functions/string-functions.md:619]:
````suggestion
```sql
SELECT 'Täglich grüßt das Murmeltier.' AS str,
       substringUTF8(str, 9),
       substringUTF8(str, 9, 5)
```
````

File: docs/en/sql-reference/functions/string-functions.md (hunk 2 above)
Comment (johnnymatthews) [inline docs/en/sql-reference/functions/string-functions.md:687]:
```suggestion
SELECT substringIndexUTF8('www.straßen-in-europa.de', '.', 2)
```

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**